scripts/package_building/utils.sh

```diff
@@ -0,0 +1,165 @@
+            lsb_release -d -s | grep -q openSUSE
+            if [[ $? -eq 0 ]]; then
+                os_vendor="openSUSE"
+            fi
+        elif [[ $os_vendor == "openSUSE project" ]]; then
+            os_vendor="openSUSE"
+        elif [[ $os_vendor =~ Red.*Hat ]]; then
+            os_vendor="Red Hat"
+        fi
+        os_codename=$(lsb_release -c -s)
+    elif [[ -r /etc/redhat-release ]]; then
+
+        #
+        # Red Hat Enterprise Linux Server release 5.5 (Tikanga)
+        # Red Hat Enterprise Linux Server release 7.0 Beta (Maipo)
+        # CentOS release 5.5 (Final)
+        # CentOS Linux release 6.0 (Final)
+        # Fedora release 16 (Verne)
+        # XenServer release 6.2.0-70446c (xenenterprise)
+        #
+        os_codename=""
+        for r in "Red Hat" CentOS Fedora XenServer; do
+            os_vendor=$r
+            if [[ -n "`grep \"$r\" /etc/redhat-release`" ]]; then
+                ver=`sed -e 's/^.* \([0-9].*\) (\(.*\)).*$/\1\|\2/' /etc/redhat-release`
+                os_codename=${ver#*|}
+                os_release=${ver%|*}
+                os_update=${os_release##*.}
+                os_release=${os_release%.*}
+                break
+            fi
+            os_vendor=""
+        done
+        os_package="rpm"
+        os_package_manager="yum"
+    elif [[ -r /etc/SuSE-release ]]; then
+        for r in openSUSE "SUSE Linux"; do
+            if [[ "$r" = "SUSE Linux" ]]; then
+                os_vendor="SUSE LINUX"
+            else
+                os_vendor=$r
+            fi
+
+            if [[ -n "`grep \"$r\" /etc/SuSE-release`" ]]; then
+                os_codename=`grep "CODENAME = " /etc/SuSE-release | sed 's:.* = ::g'`
+                os_release=`grep "VERSION = " /etc/SuSE-release | sed 's:.* = ::g'`
+                os_update=`grep "PATCHLEVEL = " /etc/SuSE-release | sed 's:.* = ::g'`
+                break
+            fi
+            os_vendor=""
+        done
+        os_package="rpm"
+        os_package_manager="TODO: suse package manager"
+
+    #
+    # If lsb_release is not installed, we should be able to detect Debian OS
+    #
+    elif [[ -f /etc/debian_version ]] && [[ $(cat /proc/version) =~ "Debian" ]]; then
+        os_vendor="Debian"
+        os_package="deb"
+        os_package_manager="apt"
+        os_codename=$(awk '/VERSION=/' /etc/os-release | sed 's/VERSION=//' | sed -r 's/\"|\(|\)//g' | awk '{print $2}')
+        os_release=$(awk '/VERSION_ID=/' /etc/os-release | sed 's/VERSION_ID=//' | sed 's/\"//g')
+        echo $os_release
+    fi
+    
+    if [[ "$os_vendor" == "Red Hat" ]] || [[ "$os_vendor" == "CentOS" ]];then
+        os_family="rhel"
+    elif [[ "$os_vendor" == "Debian" ]] || [[ "$os_vendor" == "Ubuntu" ]];then
+        os_family="debian"
+    elif [[ "$os_vendor" == "openSUSE" ]];then
+        os_family="suse"
+    fi
+    
+    echo "os_FAMILY=$os_family os_VENDOR=$os_vendor os_RELEASE=$os_release os_UPDATE=$os_update \
```

Comment:
here -- return an array with no names and make a declarative statement when you use them which is which